src/main/java/com/shaft/database/DBActions.java

```diff
@@ -0,0 +1,190 @@
+			message = message + " With the following test data [" + testData + "].";
+		}
+		ReportManager.log(message);
+		if (log != null) {
+			ReportManager.attachAsStep("Failed", "DB Connection Log", log);
+		}
+		Assert.fail(message);
+	}
+
+	private void failAction(String actionName, String testData) {
+		failAction(actionName, testData, null);
+	}
+
+	/////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
+	//////////////////////////////////// [Public] Core Database Actions
+	/////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
+
+	/**
+	 * Setup connection to database
+	 * 
+	 * @throws ClassNotFoundException
+	 * @throws SQLException
+	 * 
+	 * 
+	 */
+	
+	private void createConnection() {
+		String connectionString = "";
+		try {
+			switch (dbType.toLowerCase().trim()) {
+
+			case ("mysql"):
+				connectionString = "jdbc:mysql://" + dbServerIP + ":" + dbPort + "/" + dbName;
+				break;
+
+			case ("sqlserver"):
+				connectionString = "jdbc:sqlserver://" + dbServerIP + ":" + dbPort + ";databaseName=" + dbName;
+				break;
+
+			case ("postgresql"):
+				connectionString = "jdbc:postgresql://" + dbServerIP + ":" + dbPort + "/" + dbName;
+				break;
+
+			default:
+				ReportManager.log("Database not supported");
+				failAction("createConnection", dbType);
+				break;
+			}
+			connection = DriverManager.getConnection(connectionString, username, password);
+			passAction("createConnection", connectionString);
+		} catch (SQLException e) {
+			ReportManager.log(e);
+			failAction("createConnection", connectionString);
+		}
+	}
+	
+	/**
+	 * prepare statement for database connection to perform execute query
+	 * 
+	 * 
+	 */
+	
+	private void createStatement() {
+		try {
+			statement = connection.createStatement();
+			passAction("createStatement", connection.toString());
+		} catch (SQLException e) {
+			ReportManager.log(e);
+			failAction("createStatement", connection.toString());
+		}
+	}
+
+	/**
+	 * execute DB query and return the result as ResultSet object
+	 * 
+	 * @param dbQuery
+	 * @return ResultSet
+	 * @throws SQLException
+	 * 
+	 * return ResultSet
+	 */
+	public ResultSet executeSelectQuery(String dbQuery) {
+		ResultSet resultSet = null;
+		try {
+			createConnection();
+			createStatement();
+			resultSet = statement.executeQuery(dbQuery);
+			passAction("executeSelectQuery", dbQuery);
+		} catch (SQLException e) {
+			ReportManager.log(e);
+			failAction("executeSelectQuery", dbQuery);
+		}
+		return resultSet;
+	}
+
+	/**
+	 * Close database connection
+	 */
+	public void closeConnection() {
+		try {
+			if (resultSet != null)
+				resultSet.close();
+			if (statement != null)
+				statement.close();
+			if (connection != null)
+				connection.close();
+			ReportManager.logDiscrete("closeConnection");
+			passAction("closeConnection", connection.toString());
```

Comment:
ReportManager.logDiscrete("closeConnection"); -> This is a log message so it should be more descriptive. For example it should say clearly that the connection has been closed successfully.
passAction("closeConnection", connection.toString()); -> This line should be removed.

Reply:
done

Reply:
but I got confused, as I understood ReportManager.logDiscrete("closeConnection") is used when method is private and this one is public (will discuss later why it is public not private)